### DISCUSSION TITLE
Is Maven repo to developer.huawei banned?

### DISCUSSION THREAD

**Discussion by jingp-ofload** [codemagic-ci-cd/codemagic-docs#2002] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

1. Install [react-native-background-geolocation](https://github.com/transistorsoft/react-native-background-geolocation/blob/master/help/INSTALL-ANDROID-AUTO.md)

encounter error messages like:
```
> Could not resolve com.squareup:javapoet:1.13.0.
         > Could not get resource 'https://developer.huawei.com/repo/com/squareup/javapoet/1.13.0/javapoet-1.13.0.pom'.
            > Could not HEAD 'https://developer.huawei.com/repo/com/squareup/javapoet/1.13.0/javapoet-1.13.0.pom'.
               > Connect to developer.huawei.com:443 [developer.huawei.com/23.61.11.28, developer.huawei.com/23.61.11.31] failed: connect timed out
   > Could not resolve commons-codec:commons-codec:1.10.
     Required by:
         project :expo-updates > androidx.room:room-compiler:2.4.2
      > Could not resolve commons-codec:commons-codec:1.10.
         > Could not get resource 'https://developer.huawei.com/repo/commons-codec/commons-codec/1.10/commons-codec-1.10.pom'.
            > Could not HEAD 'https://developer.huawei.com/repo/commons-codec/commons-codec/1.10/commons-codec-1.10.pom'.
               > Connect to developer.huawei.com:443 [developer.huawei.com/23.61.11.28, developer.huawei.com/23.61.11.31] failed: connect timed out
```

### Expected results

Buil pass.

### Actual results

Build fails by time out.

**Comment by jingp-ofload** [comment]
The react-native-background-geolocation requires 
```
allprojects {
    repositories {
         .
         .
         .
+       maven {url 'https://developer.huawei.com/repo/'}
    }
} 
```
is the reason. I would like to know if codemagic banned this repo or not.

**Reply by dtrdic** [reply, marked as answer]
hey @jingp-ofload, Codemagic is not blocking the repository you mentioned. Also, when I attempt to access https://developer.huawei.com/repo/ through a web browser, I encounter an error. .. maybe this repo is deprecated or no longer available?

**Reply by jingp-ofload** [reply]
@dtrdic After I put the hms license code in `app.config.js` the build passed. Cheers.
